nixos/modules/services/web-apps/immich.nix

```diff
@@ -0,0 +1,247 @@
+
+        To set a database password set this to a file containing:
+        ```
+        DB_PASSWORD=<pass>
+        ```
+      '';
+    };
+    host = mkOption {
+      type = types.str;
+      default = "localhost";
+      description = "The host that immich will listen on.";
+    };
+    port = mkOption {
+      type = types.port;
+      default = 3001;
+      description = "The port that immich will listen on.";
+    };
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Whether to open the immich port in the firewall";
+    };
+    user = mkOption {
+      type = types.str;
+      default = "immich";
+      description = "The user immich should run as.";
+    };
+    group = mkOption {
+      type = types.str;
+      default = "immich";
+      description = "The group immich should run as.";
+    };
+
+    machine-learning = {
+      enable = mkEnableOption "immich-machine-learning" // { default = true; };
+      environment = mkOption {
+        type = types.submodule {
+          freeformType = types.attrsOf types.str;
+        };
+        default = {};
+        example = {
+          MACHINE_LEARNING_MODEL_TTL = "600";
+        };
+        description = ''
+          Extra configuration environment variables. Refer to the [documentation](https://immich.app/docs/install/environment-variables) for options tagged with 'machine-learning'.
+        '';
+      };
+    };
+
+    database = {
+      enable = mkEnableOption "postgresql" // { default = true; };
+      createDB = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether to create the database for immich automatically.";
+      };
+      name = mkOption {
+        type = types.str;
+        default = "immich";
+        description = "The name of the immich database.";
+      };
+      host = mkOption {
+        type = types.str;
+        default = "/run/postgresql";
+        example = "127.0.0.1";
+        description = "Hostname or address of the postgresql server. If an absolute path is given here, it will be interpreted as a unix socket path.";
+      };
+      user = mkOption {
+        type = types.str;
+        default = "immich";
+        description = "The database user for immich.";
+      };
+      setupPgvectors = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether to add the pgvecto.rs extension (required to run immich) to postgresql.";
+      };
+    };
+    redis = {
+      enable = mkEnableOption "immich redis server" // { default = true; };
+      host = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = "The host that redis will listen on.";
+      };
+      serverName = mkOption {
+        type = types.str;
+        default = "immich";
+        description = "The server name of the immich redis instance.";
+      };
+      port = mkOption {
+        type = types.port;
+        default = 6379;
```

Comment:
Would it make more sense to use a socket for Redis like you do for Postgres? That’s what the Nextcloud module does 
https://github.com/NixOS/nixpkgs/blob/e2dd4e18cc1c7314e24154331bae07df76eb582f/nixos/modules/services/web-apps/nextcloud.nix

Reply:
Looks like immich does not like that

Reply:
Bummer, thanks for checking it out though.